### PR TITLE
Replace old CRW datasync walkthrough with the latest one

### PR DESCRIPTION
**Pull request by mmusil**
### Motivation

Replacing old walkthrough with a new one without Code Ready as it has proved to be ustable. This PR was created as a follow up on #33 .

### CODE REVIEW

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 1]
```diff
@@ -54,61 +51,157 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
-Wait until you see that all three pods have the *Running* status. Once the pods are running, you can use the GraphQL playground to interact with the server API as described in the next step.
+DataSync will allow your application to recieve live updates thanks to GraphQL subscriptions and
+operate independently of network connection. Developers can create GrapQL types as model
+and generate underlying backend that will work out of the box with the RHMI services like SSO or AMQ Online
+
+[time=30]
+=== Setting your DataSync project using DataSync Starter
+
+. Install yarn `npm install -g yarn`
+. Clone git repository: `git clone https://github.com/aerogear/datasync-starter#walkthrough`
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc:68]: Could you do this with the master. I will push master to walkthrough once we happy with this WT

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 2]
```diff
@@ -54,61 +51,157 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
+. Go to cloned repository: `cd datasync-starter`
+. Install dependencies `yarn install`
+
+=== Creating your DataSync model
+
+DataSync model offers capability to generate underlying GraphQL API and database infrastructure.
+Starter template by default will use MongoDB database as datasource for storing all your data.
+
+. In your project go to `./model/task.graphql` file.
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc:77]: https://github.com/aerogear/datasync-starter/tree/master/server/model

```suggestion
. In your project go to `./server/model/task.graphql` file.
```

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 3]
```diff
@@ -54,61 +51,157 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
+This file contains GraphQL compatible schema
+. Please add `@datasync` annotation to the model right bellow `@model` annotation
+Those annotations can be used to control various behaviour of datasync.
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc:80]:
```suggestion
This annotation will enable data synchornization for this particular object. You can read more about data synchronization here https://graphback.dev/docs/datasync/soft-delete
```

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 4]
```diff
@@ -54,61 +51,157 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
+`@model` will create standard data access methods, while `@datasync` is going to provide data synchronization capabilities.
+. Models can be changed by adding new types or fields. Please add new field by creating new line inside `Task` type
+and `address: String`
+. Your task model should look as follows
+----
+"""
+@model
+@datasync
+"""
+type Task {
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc:90]: This is outdated 
https://github.com/aerogear/datasync-starter/blob/master/server/model/task.graphql#L5-L20

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 5]
```diff
@@ -54,61 +51,157 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
+- `.graphqlrc.yml`
+
+. Review `.graphqlrc.yml` file. This file contains configuration for your data access methods
+and available plugins that will be used for source code generation
+. Please make sure that all fields in `crud` section are enabled
+. Execute graphback cli command to generate source code:
+`yarn graphback generate`
+. Review `./server/src/schema/schema.qraphql`.
+This file contains your model along with access methods that were enabled in configuration.
+. Review generated resolver files in `./server/src/resolvers/resolvers.ts`
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc:116]:
```suggestion
```

Remove this. We no longer generate resolves by default

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 6]
```diff
@@ -54,61 +51,157 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
+This file contains methods used to fetch data. Each individual method will use
+preconfigured MongoDataProvider. Developers can point resolvers to any datasource.
+Currently Postres and MongoDB are supported.
+. Review your `./client/src/graphql/` folder containing client side queries for your data.
+
+=== Running DataSync client and server applications
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc:122]:
```suggestion
=== Running DataSync Client and Server applications
```

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 7]
```diff
@@ -54,61 +51,157 @@ image::images/arch.png[architecture, role="integr8ly-img-responsive"]
...
     title,
     description,
+    address,
     id
   }
 }
 
 mutation createTask {
-  createTask(title: "complete the walkthrough", description: "complete the GraphQL walkthrough") {
+  createTask(title: "complete the walkthrough", description: "complete the GraphQL walkthrough", address: "NA") {
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc:177]: This query is oudated.

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 8]
```diff
@@ -452,13 +343,64 @@ Conflicting description from laptop
...
+== Add authentication and authorization to the Data Sync application using Red Hat SSO
+
+In this task, we will configure both the frontend and the backend of the
+{data-sync-starter} with the {customer-sso-name}.
+
+DataSync starter has authentication and autorization enabled out of the box.
+Developers need to configure server and client application to use their keycloak instance
+and add required authorization rules to their model.
+
+=== Add authorization rule for Task deletion
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc]: This is outdated. 

We can just say:

Please review auth file (https://github.com/aerogear/datasync-starter/blob/master/server/src/config/auth.ts#L8). We can see that only admin can delete task. 
For more information check https://graphback.dev/docs/authentication/keycloak

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc [hunk 9]
```diff
@@ -452,13 +343,64 @@ Conflicting description from laptop
...
+Schema contains mutations section that is responsible for data modifications
+. In mutation section find `deleteTask(input: TaskInput): Task!`
+. Add GraphQL Directive on top of it `@hasRole(role: "admin")`
+This will only allow deletion for admin users.
+Roles can be also applied in generation process by utilizing graphback plugin
+. This directive is already defined in {data-sync-starter} and can be also applied
+to any new mutation or query created by users.
+We going to verify this directive in next steps
+
+=== Configuring Authentication for Keycloak (SSO) (Local setup)
```
Comment (wtrocki) [inline walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc]: I'm not sure if we can do keycloak locally considered that there is one on RHMI. We need to copy paste entire instruction for keycloak from #39 that will be up to date

**Comment by wtrocki** [comment]
Amazing work @mmusil. This works on the walkthrough branch but this branch is also outdated due to our recent release and the fact that local guide was build 3 months ago. If we update this to still connect to keycloak that runs on RHMI (we can later extend to deploy this application to openshift and try it etc.)

It is very solid base that we will work on regularly and has very little risk due to limited integration to RHMI services.

File: walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc (hunk 9 above)
Comment (mmusil) [reply walkthroughs/5-adding-data-sync-graphql/walkthrough.adoc]: Did you mean copy-paste it from #33? :)

**Pull request merged**